
# Poyecto 1 programación lineal 
### Optimización. Maximizar utilidades
Una compañía fabrica automóviles, camionetas, motos y camiones. Cada uno le da una utilidad de $\$30,000$ pesos, $\$40,000$ pesos, $\$15,000$ y $\$50,000$ respectivamente. 

Al mes la cantidad máxima que se puede producir de carros es de 400.  De camionetas máximo 350, motos 250 y camiones 100. La fabrica solo puede trabajar 720 horas al mes, la producción de cada vehículo en cuestión del tiempo es de una hora. 
La compañía quiere saber cuantas vehículos fabricar al mes para alcanzar su utilidad máxima. Suponiendo que todo lo vende.

---

In [64]:
# Importar librerías 

import scipy.optimize as opt
import numpy as np 

In [65]:
# Definir matrices 

C=np.array([-30000,-40000,-15000,-50000])
A=np.array([[1,0,0,0],[0,1,0,0],[0,0,1,0],[0,0,0,1],[1,1,1,1]])
B=np.array([400,350,250,100,720])


sol=opt.linprog(C,A,B,method='simplex')
sol.x

array([270., 350.,   0., 100.])

Para alcanzar la máxima utilidad se necesitan fabricar 270 autos, 350 camionetas, 0 motos y 100 camiones. 

In [66]:
# Mostrar la utilidad máxima
(-1)*(sol.fun)

27100000.0

La ganancia máxima es de $\$27,100,000$ pesos al mes